.github/workflows/test.yml

```diff
@@ -20,32 +17,37 @@ jobs:
       - name: Checkout github repo (+ download lfs dependencies)
         uses: actions/checkout@v2
         with:
-          lfs: true
           fetch-depth: 0
 
-      - name: Install miniconda and environment
-        uses: conda-incubator/setup-miniconda@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
         with:
-          miniconda-version: "latest"
-          activate-environment: btk_py37
           python-version: ${{ matrix.python-version }}
 
-      - name: Install conda dependencies
-        shell: bash -l {0} # setup and activate conda environment.
+      - name: Install Poetry
+        run: |
+          pip install --upgrade pip
+          pip install poetry
+
+      - name: Poetry cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/pypoetry
+          key: ${{ runner.os }}-${{ hashFiles('./poetry.lock') }}
+
+      # galsim needs this.
+      - name: Instal FFTW + Eigen
         run: |
-          conda install fftw
-          conda install -c conda-forge eigen
-          conda install -c conda-forge galsim
-          conda install tensorflow==1.15
-          conda install -c conda-forge poetry
+          sudo apt-get install libfftw3-dev libeigen3-dev
 
       - name: Install poetry dependencies
-        shell: bash -l {0}
         run: |
+          poetry run pip install numpy Cython
+          poetry run pip install "eigency>=1.77"
           poetry install --extras "galsim-hub"
```

Comment:
I managed to get `poetry` up and running in the workflow again (no more conda!) . The only problem is `galsim` which fails if I don't install `numpy, Cython, eigency` before everything else. I'd be very interested in hearing a solution to this (but not urgent)